controllers/release/adapter.go

```diff
@@ -478,3 +490,42 @@ func (a *Adapter) syncResources() error {
 
 	return a.syncer.SyncSnapshot(snapshot, releasePlanAdmission.Namespace)
 }
+
+// trackReleaseAttribution updates the status of the Release being processed with the proper attribution author.
+func (a *Adapter) trackReleaseAttribution(releasePlan *v1alpha1.ReleasePlan) error {
+	if a.release.IsAttributed() {
+		return nil
+	}
+
+	var author string
+	patch := client.MergeFrom(a.release.DeepCopy())
+
+	if a.release.Labels[metadata.AutomatedLabel] == "true" {
+		if !a.release.IsAutomated() {
```

Comment:
This function is about properly setting the Attribution.Author field in the release status. Why I duplicate the retrieving of the automated label in both registerAttributionData and validateAuthor instead of just do it all here?

Reply:
Because you are mixing the logic. Also, I'm not saying that you have to duplicate the check. A function to set the author should not care about raising an error which is used to decide how the reconciliation should continue. As we discussed offline, one function to validate and another one to register the data.